src/vaults/Vault.sol

```diff
@@ -125,8 +128,23 @@ contract Vault is AffineVault, ERC4626Upgradeable, PausableUpgradeable, Detailed
     {
         _liquidate(assets);
 
-        // Slippage during liquidation means we might get less than `assets` amount of `_asset`
-        assets = Math.min(_asset.balanceOf(address(this)), assets);
+        // If user tries to withdraw too much, register request on LockedWithdrawalEscrow
+        uint256 currAssets = _asset.balanceOf(address(this));
+        if (currAssets < assets) {
+            uint256 sharesToLock = _convertToShares(assets - currAssets, MathUpgradeable.Rounding.Up);
+            // Lock users shares in escrow
+            _transfer({from: owner, to: address(debtEscrow), amount: sharesToLock});
+
+            // Register withdrawal request
+            debtEscrow.registerWithdrawalRequest(_msgSender(), sharesToLock);
```

Comment:
we need to register debt of `assets - currAssets`. cause debt token = shares* price.